### PR TITLE
Fix the compatibility workflows

### PR DESCRIPTION
**Pull request by pradeepbn**
Descriptions of the changes in this PR:
- Fix the compatibility workflows for java11 and java8
- Fixes the dependencies in the tests

### Motivation

Compatibility checks workflow wasn't working because there the github workflow.yaml had syntax issues. Also, since it was enabled 1st time, there were build issues with the tests.

### CODE REVIEW

**Comment by pkumar-singh** [comment]
rerun failure checks

**Review by pkumar-singh — APPROVED** [review]

**Review by pkumar-singh — APPROVED** [review]

**Review by Vanlightly — APPROVED** [review]
LGTM

**Comment by Vanlightly** [comment]
@eolivelli could PTAL? If all is good would be good to merge.

File: tests/shaded/distributedlog-core-shaded-test/src/test/java/org/apache/bookkeeper/tests/shaded/DistributedLogCoreShadedJarTest.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public void testProtobufIsShaded() throws Exception {
 
     @Test
     public void testProtobufShadedPath() throws Exception {
-        Class.forName("dlshade.com.google.protobuf.Message");
+        Class.forName("org.apache.bookkeeper.shaded.com.google.protobuf.Message");
```
Comment (nicoloboschi) [inline tests/shaded/distributedlog-core-shaded-test/src/test/java/org/apache/bookkeeper/tests/shaded/DistributedLogCoreShadedJarTest.java:62]: Why this? I think there is a duplicate rule for com.google in the build.gradle

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
Great work, I think we should not change the current tests about shaded DL classpath

File: tests/shaded/distributedlog-core-shaded-test/src/test/java/org/apache/bookkeeper/tests/shaded/DistributedLogCoreShadedJarTest.java (hunk 1 above)
Comment (pradeepbn) [reply tests/shaded/distributedlog-core-shaded-test/src/test/java/org/apache/bookkeeper/tests/shaded/DistributedLogCoreShadedJarTest.java]: Good catch- thanks. Its fixed now.

**Review by eolivelli — APPROVED** [review]

**Comment by eolivelli** [comment]
Tests didn't pass.
I have restarted them

**Pull request merged**